### PR TITLE
new lint avoid_single_cascade_in_expression_statements

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: lib/src/rules/avoid_single_cascade_in_expression_statements.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+class Visitor extends SimpleAstVisitor {
+  Visitor(this.rule);
+
+  final LintRule rule;
+
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    if (node.cascadeSections.length == 1 && node.parent is ExpressionStatement) {
```
Comment (bwilkerson) [inline lib/src/rules/avoid_single_cascade_in_expression_statements.dart]: I'm not sure why this is limited to the statement level. Isn't it just as bad to have `f(o..m())`? What's the use case for limiting this?

Comment (a14n) [reply lib/src/rules/avoid_single_cascade_in_expression_statements.dart]: `f(o..m())` will call `f` with `o` but `f(o.m())` will call `f` with the result of `o.m()`.

Comment (bwilkerson) [reply lib/src/rules/avoid_single_cascade_in_expression_statements.dart]: Oh, of course. I should have seen that.

**Comment by a14n** [comment]
CI fixed. It can now be merged.

**Comment by a14n** [comment]
Thanks

**Pull request merged**